identity-iota/src/credential/errors.rs

```diff
@@ -0,0 +1,127 @@
+                                                                 * here? */
+  },
+
+  /// Indicates an attempt to verify a presentation's signature with a resolved DID document not corresponding to the
+  /// URL of the presentation's holder property.
+  #[error("the provided holder document does not correspond to the presentation's holder property")]
+  #[non_exhaustive]
+  IncompatibleHolderDocument,
+
+  /// Indicates that the presentation's signature could not be verified using the holder's DID Document.
+  #[error("could not verify the holder's signature")]
+  #[non_exhaustive]
+  HolderProof {
+    source: Box<dyn std::error::Error + Send + Sync + 'static>, /* Todo: Would it be better to use a specific type
+                                                                 * here? */
+  },
+  /// Indicates that the structure of the [Credential](identity_credential::credential::Credential) is not semantically
+  /// correct.
+  #[error("the credential's structure is not semantically correct")]
+  CredentialStructure(#[source] identity_credential::Error),
+  /// Indicates that the structure of the [Presentation](identity_credential::presentation::Presentation) is not
+  /// semantically correct.
+  #[error("the presentation's structure is not semantically correct")]
+  PresentationStructure(#[source] identity_credential::Error),
+  /// Indicates that the relationship between the presentation holder and one of the credential subjects is not valid.
+  #[error("expected holder = subject of the credential at position {credential_position}")]
+  #[non_exhaustive]
+  InvalidHolderSubjectRelationship { credential_position: usize },
+  /// Indicates that the presentation does not have a holder.
+  #[error("the presentation has an empty holder property")]
+  MissingPresentationHolder,
+}
+
+// Todo: Consider implementing Error for OneOrMany<E: std::error::Error> to avoid wrapping it in
+// AccumulatedCredentialValidationError
+#[derive(Debug)]
+/// An error caused by a failure to validate a Credential.  
+pub struct AccumulatedCredentialValidationError {
+  pub validation_errors: OneOrMany<ValidationError>,
+}
+
+impl Display for AccumulatedCredentialValidationError {
+  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    // intersperse might become available in the standard library soon: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.intersperse
+    let detailed_information: String = itertools::intersperse(
+      self.validation_errors.iter().map(|err| err.to_string()),
+      ", ".to_string(),
+    )
+    .collect();
+    write!(f, "The following errors occurred: {}", detailed_information)
+  }
+}
+
+impl std::error::Error for AccumulatedCredentialValidationError {}
+
+#[derive(Debug)]
+/// An error caused by a failure to validate a Presentation.
+pub struct AccumulatedPresentationValidationError {
```

Comment:
Suggestion: is `CompoundPresentationValidationError` any better than `Accumulated`? Slightly shorter I guess.